src/deploy/campaign/environments.ts

```diff
@@ -117,8 +119,56 @@ export const getConfig = (
     rootPriceConfig: priceConfig,
     zeroVaultAddress: process.env.ZERO_VAULT_ADDRESS ? process.env.ZERO_VAULT_ADDRESS : zeroVault.address,
     mockMeowToken: process.env.MOCK_MEOW_TOKEN ? !!process.env.MOCK_MEOW_TOKEN : true,
-    stakingTokenAddress: process.env.STAKING_TOKEN_ADDRESS ? process.env.STAKING_TOKEN_ADDRESS : MEOW_TOKEN
+    stakingTokenAddress: process.env.STAKING_TOKEN_ADDRESS ? process.env.STAKING_TOKEN_ADDRESS : MEOW_TOKEN,
   };
 
+  // Will throw an error based on any invalid setup, given the `ENV_LEVEL` set
+  validate(config);
+
   return config;
 };
+
+// For testing the behaviour when we manipulate, we have an optional "env" string param
+export const validate = (config : IDeployCampaignConfig, env ?: string, mongoUri ?: string) => {
+  let envLevel;
+
+  if (env) {
+    envLevel = env;
+  } else {
+    envLevel = process.env.ENV_LEVEL;
+  }
+
+  if (envLevel === "dev") return; // No validation needed for dev
+
+  if (!mongoUri) mongoUri = process.env.MONGO_URI ? process.env.MONGO_URI : DEFAULT_MONGO_URI;
+
+  // Mainnet or testnet
+  if (envLevel === "prod" || envLevel === "test") {
+    requires(!config.mockMeowToken, "Cannot mock MEOW token in production");
+    requires(config.stakingTokenAddress === MEOW_TOKEN, "Must use MEOW token in production");
+    requires(validatePrice(config.rootPriceConfig), "Must use a valid price configuration");
+    requires(!mongoUri.includes("localhost"), "Cannot use local mongo URI in production");
+  }
+
+  // If we reach this code, there is an env variable but it's not valid.
+  throw new Error("Invalid environment value. Must set env to one of `dev`, `test`, or `prod`");
```

Comment:
use the constants you created for this and the above errors. the same constants you use in tests